llvm/lib/Transforms/Vectorize/VPlanValue.h

```diff
@@ -121,18 +121,14 @@ class VPValue {
 
   /// Remove a single \p User from the list of users.
   void removeUser(VPUser &User) {
-    bool Found = false;
     // The same user can be added multiple times, e.g. because the same VPValue
     // is used twice by the same VPUser. Remove a single one.
-    erase_if(Users, [&User, &Found](VPUser *Other) {
-      if (Found)
-        return false;
-      if (Other == &User) {
-        Found = true;
-        return true;
+    for (const auto &U : Users) {
```

Comment:
Could we do

```
auto I = llvm::find(Users, User);
if (I != Users.end())
  Users.erase(I);
```

Or its expected to always find it, you could do
`Users.erase(llvm::find(Users, User))`